src/components/v2/deployment-history/ea/EADeploymentHistory.component.tsx

```diff
@@ -34,19 +49,57 @@ function ExternalAppDeploymentHistory({appId}) {
             });
     }, []);
 
-    function changeDeployment(index: number){
-        if(selectedDeploymentHistoryIndex == index){
+    function changeDeployment(index: number) {
+        if (selectedDeploymentHistoryIndex == index) {
             return;
         }
 
         setSelectedDeploymentHistoryIndex(index);
+
+        // Resetting the deployment tab selection, loading & error states on version change.
+        setSelectedDeploymentTabIndex(0);
```

Comment:
Purpose of this line?

Reply:
For resetting deployment tab selection after changing the version. As it was maintaining the tab state from old version selection, displaying YAML of the old version.